lib/src/builder/parsing/version.dart

```diff
@@ -0,0 +1,207 @@
+// Copyright 2020 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:meta/meta.dart';
+
+import 'members.dart';
+
+/// The confidence levels related to boilerplate [Version]s.
+///
+/// As the builder detects members, it generates a confidence based on the
+/// implementation details of the member. This confidence score is then averaged
+/// over all of the grouped members (factory, props, state, component, etc).
+///
+/// Related: [resolveVersion]
+@sealed
+abstract class Confidence {
+  /// Not possible to be considered a given version.
+  static const none = 0;
+
+  /// Unlikely to be considered a given version.
+  static const unlikely = 1;
+
+  /// Equally likely and unlikely to be considered a given version.
+  ///
+  /// Consuming code should only consider values strictly greater than this value
+  /// as confident enough to perform code generation:
+  ///
+  ///    - `confidence <= neutral`: do not perform code generation
+  ///    - `confidence > neutral`: perform code generation
+  static const neutral = 10;
```

Comment:
Looking at this code, I'm not clear on where these numbers come from? Is that somewhere else that I'm missing?

Reply:
These numbers are kind of arbitrary to allow certain combinations of confidences yield certain results as well; I can update the doc comments to indicate that.

e.g.,

`average([neutral, likely, unlikely]) > neutral`
`average([neutral, neutral, neutral, neutral, likely]) > neutral`
`average([neutral, neutral, neutral, neutral, unlikely]) < neutral`

So I can also do that!

Reply:
ah I see, thanks!

Reply:
See https://github.com/Workiva/over_react/pull/481/commits/984eddbe238b4e356467db27664ca5c9297b8939